content.go

```diff
@@ -42,3 +47,53 @@ func Tag(ctx context.Context, target Target, src, dst string) error {
 	}
 	return target.Tag(ctx, desc, dst)
 }
+
+// DefaultResolveOptions provides the default ResolveOptions.
+var DefaultResolveOptions ResolveOptions
+
+// ResolveOptions contains parameters for oras.Resolve.
+type ResolveOptions struct {
+	// TargetPlatform ensures the resolved content matches the target platform
+	// if the node is a manifest, or selects the first resolved content that
+	// matches the target platform if the node is a manifest list.
+	TargetPlatform *ocispec.Platform
+}
+
+// Resolve resolves a descriptor with provided reference from the target.
+func Resolve(ctx context.Context, target Target, ref string, opts ResolveOptions) (ocispec.Descriptor, error) {
+	if opts.TargetPlatform == nil {
+		return target.Resolve(ctx, ref)
+	}
+
+	refFetcher, ok := target.(registry.ReferenceFetcher)
+	if ok {
+		desc, rc, err := refFetcher.FetchReference(ctx, ref)
+		if err != nil {
+			return ocispec.Descriptor{}, err
+		}
+		defer rc.Close()
+
+		switch desc.MediaType {
+		case docker.MediaTypeManifestList, ocispec.MediaTypeImageIndex,
+			docker.MediaTypeManifest, ocispec.MediaTypeImageManifest:
+			store := cas.NewMemory()
+			err = store.Push(ctx, desc, rc)
+			if err != nil {
+				return ocispec.Descriptor{}, err
+			}
+
+			proxy := cas.NewProxy(target, store)
+			proxy.StopCaching = true
```

Comment:
This is good!  Better add some comment for this block like what is optimized with the proxy.

Reply:
Added. 👌